ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStream.java

```diff
@@ -37,34 +39,68 @@
 
 public class TestDataStream extends BaseTest {
 
-  private RaftPeer[] peers;
+  private List<RaftPeer> peers;
   private RaftProperties properties;
-  private DataStreamServerImpl server;
+  private List<DataStreamServerImpl> servers;
   private DataStreamClientImpl client;
   private List<CompletableFuture<DataStreamReply>> futures = new ArrayList<>();
 
   public void setupServer(){
-    server = new DataStreamServerImpl(peers[0], properties, null);
-    server.getServerRpc().startServer();
+    servers = new ArrayList<>(peers.size());
+    // start stream servers on raft peers.
+    for (int i = 0; i < peers.size(); i++) {
+      if (i == 0) {
+        // only the first server routes requests to peers.
+        List<RaftPeer> otherPeers = new ArrayList<>(peers);
+        otherPeers.remove(peers.get(i));
+        DataStreamServerImpl streamServer =
+            new DataStreamServerImpl(peers.get(i), properties, null, otherPeers);
+        servers.add(streamServer);
+        streamServer.getServerRpc().startServer();
+      } else {
+        DataStreamServerImpl streamServer =
+            new DataStreamServerImpl(peers.get(i), properties, null);
+        servers.add(streamServer);
+        streamServer.getServerRpc().startServer();
+      }
```

Comment:
DataStreamServerImpl streamServer;
      if (i == 0) {
        // only the first server routes requests to peers.
        List<RaftPeer> otherPeers = new ArrayList<>(peers);
        otherPeers.remove(peers.get(i));
        streamServer =
            new DataStreamServerImpl(peers.get(i), properties, null, otherPeers);
      } else {
        streamServer =
            new DataStreamServerImpl(peers.get(i), properties, null);
      }
        servers.add(streamServer);
        streamServer.getServerRpc().startServer();

Reply:
will address this comment later.

Reply:
has not been addressed

Reply:
oops. Have pushed another commit to address this one.